### PR TITLE
Audit logs

### PR DESCRIPTION
**Pull request by azhavoro**
### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into the [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

**Comment by bsekachev** [comment]
Probably something does not work for ``job`` query parameter. 

image

I would suggest to rename ``organization`` parameter to ``org`` (like for other REST API endpoints)

**Comment by bsekachev** [comment]
For load:job event we have job and task in payload (as task id and job id).  For other events we have them like a part of the body. I believe they should be in the body also for load:job event. 

image

Organization is always null, but I work in an organization. This information also can be received from POST request query parameter. This way it would work also for server-side create events for resources (like tasks, projects, etc).

File: cvat/apps/events/views.py
Comment (bsekachev) [inline cvat/apps/events/views.py:3]: Not Intel anymore (: 
The same in multiple files.

File: cvat/apps/events/export.py
Comment (bsekachev) [inline cvat/apps/events/export.py:92]: ``jid`` or ``job``?

File: .github/workflows/main.yml
Comment (nmanovic) [inline .github/workflows/main.yml:56]: @azhavoro , should we revert the change?

File: components/analytics/clickhouse/init.sh
Comment (nmanovic) [inline components/analytics/clickhouse/init.sh:11]: Probably it is necessary to check that CLICKHOUSE_USER value without spaces and special symbols.

File: cvat-core/src/enums.ts
Comment (nmanovic) [inline cvat-core/src/enums.ts:78]: I don't see `create:task`, `create:project`, `sort:task`, `filter:task`. Are we going to handle such events on UI?

File: cvat-core/src/log.ts
Comment (nmanovic) [inline cvat-core/src/log.ts:11]: EventLogger

File: cvat-core/src/log.ts
Comment (nmanovic) [inline cvat-core/src/log.ts:151]: Do we really need a separate class? I believe it can be decided based on LogType (maybe EventType) value.

File: cvat-core/src/logger-storage.ts
Comment (nmanovic) [inline cvat-core/src/logger-storage.ts:187]: send_timestamp -> timestamp

File: cvat-core/src/server-proxy.ts
Comment (nmanovic) [inline cvat-core/src/server-proxy.ts:1888]: Better to rename the function as well. In general we don't send logs now. We send events which can be exported and processed.

File: docker-compose.yml
Comment (nmanovic) [inline docker-compose.yml:328]: cvat_logs -> cvat_events. Let's rename all volumes (logs -> events)
cvat_events_db

File: site/content/en/docs/administration/advanced/analytics.md
Comment (nmanovic) [inline site/content/en/docs/administration/advanced/analytics.md:16]: logs -> events

File: docker-compose.yml
Comment (nmanovic) [inline docker-compose.yml:213]: CVAT_EVENT_VIEWER_HOST

File: docker-compose.yml
Comment (nmanovic) [inline docker-compose.yml:239]: Need to have a well-defined pattern to name services. Now some of them have "cvat_" prefix, and some of them don't.  I don't see the logic.

File: docker-compose.dev.yml
Comment (nmanovic) [inline docker-compose.dev.yml:72]: `docker-compose -f docker-compose.yml -f docker-compose.dev.yml   up -d --build` doesn't work for me anymore because I have redis on my computer. What is a reason to export the port?

Comment (nmanovic) [reply docker-compose.dev.yml:72]: OK. Probably it is fine. But I don't like that in the dev file we open so many ports. It can lead to conflicts.
I have started thinking that we need analytics.yml for all new services.

Comment (nmanovic) [reply docker-compose.dev.yml:72]: I will recommend not opening the port, because if a user has running redis on the host, it will lead to problems.

**Review by nmanovic — COMMENTED** [review]
If I press on the Grafana icon, I can see the login page. Is it possible to disable?

**Review by nmanovic — COMMENTED** [review]
Need to have a solution for home dashboard. The current solution isn't user-friendly. 

https://github.com/grafana/grafana/issues/10266#issuecomment-673519617

**Comment by nmanovic** [comment]
@azhavoro , I can not see payload of events
image

**Comment by nmanovic** [comment]
@azhavoro , I don't understand why create:annotations event has obj_name = job, obj_id = 1, etc.
image

**Comment by nmanovic** [comment]
I don't think we need send:task_info from the client. I don't understand why it provides "working time". Full statistics about the job should be extracted from the server audit logs.  

```
{"frame count": 124, "track count": 1, "object count": 1, "box count": 1, "polygon count": 0, "polyline count": 0, "points count": 0, "cuboids count": 0, "tag count": 0, "client_id": "581764", "is_active": true, "working_time": 470}
```

**Comment by nmanovic** [comment]
What problems I see below:

- Server audit logs doesn't have create:job event. We should have all events. We should be able to track every change of the server state.
- I don't press fit image
- It is unclear which object I drew
- Payload for draw:object has working time.

image

**Comment by nmanovic** [comment]
I will recommend don't put update:annotations and delete:annotations events if they don't have any value in the context.
Also, instead of create:annotations, I would put create:bbox and into the count field I would put the number of bounding boxes. 

image

**Comment by nmanovic** [comment]
need to understand how to use obj_name, obj_id, obj_val fields properly. They should not duplicate information which we already have.

**Comment by nmanovic** [comment]
For a user we should have both: id and its username

**Comment by nmanovic** [comment]
update:assignee should have obj_name = user, obj_id = assignee_id, obj_val = username (or something like that)

image

**Comment by nmanovic** [comment]
When a task is deleted, we don't have events delete:job in audit logs.

**Comment by nmanovic** [comment]
I don't see any information about a created organization

**Comment by nmanovic** [comment]
SQL query for "all events" should be something like

```
SELECT * 
FROM events 
WHERE $__timeFilter(timestamp)
ORDER BY timestamp DESC
```

remove unnecessary filters, also it is important to have DESC. Because otherwise after update you will see the first page every time.

File: cvat/apps/engine/log.py
Comment (nmanovic) [inline cvat/apps/engine/log.py:171]: Кстати, я не думаю, что нам нужно писать логи в файлы, как мы это делали (см выше). Теперь все логи должны быть в ClickHouse

File: cvat/apps/engine/mixins.py
Comment (nmanovic) [inline cvat/apps/engine/mixins.py:343]: We already have multiple problems in WebHooks app due to own signals. Please look at the issue: https://github.com/opencv/cvat/issues/5631

Use Django DB signals only! They should be enough to track how the server state is changed in most cases.

File: cvat/apps/engine/task.py
Comment (nmanovic) [inline cvat/apps/engine/task.py:166]: Use Django DB signals only: https://docs.djangoproject.com/en/4.1/ref/signals/#module-django.db.models.signals

File: cvat/apps/engine/views.py
Comment (nmanovic) [inline cvat/apps/engine/views.py:1491]: Don't use own signals

File: cvat/apps/events/apps.py
Comment (nmanovic) [inline cvat/apps/events/apps.py:8]: Need to rename the app. Think about consistency in the code.

File: cvat/apps/events/formatter.py
Comment (nmanovic) [inline cvat/apps/events/formatter.py:7]: class EventFormatter

Could you please write a comment about why we need it? It looks dummy.

File: cvat/apps/iam/permissions.py
Comment (nmanovic) [inline cvat/apps/iam/permissions.py:339]: Probably DUMP_EVENTS = 'dump:events' is a better name.

File: cvat/apps/iam/rules/server.rego
Comment (nmanovic) [inline cvat/apps/iam/rules/server.rego:43]: Why business?

Which code is responsible for filtration? Can I get all events? IAM should decide which events I can get. For example, if I a maintainer of an organization, I can see all events of the organization. At the same time, if I'm a worker, I can see only my own events.

File: cvat/requirements/base.txt
Comment (nmanovic) [inline cvat/requirements/base.txt:56]: 0.5.7

File: cvat/apps/events/views.py
Comment (nmanovic) [inline cvat/apps/events/views.py:67]: Why do we need the action in EventsViewSet? Why don't use 'create' and 'list' methods?

File: cvat/apps/events/views.py
Comment (nmanovic) [inline cvat/apps/events/views.py:123]: Do we really want to have a separate endpoint for exceptions? From my perspective is a regular event.

File: cvat/apps/events/views.py
Comment (nmanovic) [inline cvat/apps/events/views.py:75]: Can we adjust timestamps inside the serializer? What are pros and cons? I believe it is a part of de-serialization logic.

File: cvat/apps/events/views.py
Comment (nmanovic) [inline cvat/apps/events/views.py:100]: I don't think that we need clogger at all. What do you think?

File: cvat/apps/events/export.py
Comment (nmanovic) [inline cvat/apps/events/export.py:99]: Can we have an exception here? Should we protect ourselves from a wrong user input?

File: cvat/apps/events/views.py
Comment (nmanovic) [reply cvat/apps/events/views.py:67]: Also, we need to think about the interface for long operations. For example, how to get the list of all operations which are in progress now? Let's say we reload UI, we don't know query_id anymore. How to download the CSV file?

File: cvat/apps/events/export.py
Comment (nmanovic) [inline cvat/apps/events/export.py:88]: From my perspective, we should implement the functionality using DjangoFilterBackend: https://www.django-rest-framework.org/api-guide/filtering/

Please work with Maxim to understand the best solution here.
Look at the PR: https://github.com/opencv/cvat/pull/5575

File: docker-compose.yml
Comment (azhavoro) [reply docker-compose.yml:255]: discussed offline

File: docker-compose.yml
Comment (azhavoro) [reply docker-compose.yml]: add `cvat_` prefix

File: docker-compose.dev.yml
Comment (azhavoro) [reply docker-compose.dev.yml]: removed

File: cvat/apps/engine/log.py
Comment (azhavoro) [reply cvat/apps/engine/log.py:139]: I perefere to keep server logs for first iteration

File: cvat/apps/iam/rules/server.rego
Comment (azhavoro) [reply cvat/apps/iam/rules/server.rego]: Implemented by @kirill-sizov

File: cvat/apps/events/views.py
Comment (azhavoro) [reply cvat/apps/events/views.py]: deleted

File: cvat/apps/events/export.py
Comment (azhavoro) [reply cvat/apps/events/export.py:89]: Discussed with Maxim: DjangoFilterBackend doesn't provide any advantages here due to there is not a django queryset.

File: cvat-ui/src/utils/controls-logger.ts
Comment (bsekachev) [inline cvat-ui/src/utils/controls-logger.ts:32]: Need to think what kind of information we are going to record about pressed buttons. Maybe  need to add descriptive class name to all the buttons. Now not everything has it and it is not always clear what button has been pressed.

File: cvat-ui/src/utils/controls-logger.ts
Comment (bsekachev) [inline cvat-ui/src/utils/controls-logger.ts:54]: For now the implementation is allright, but in the future we probably need to generalize it somehow

**Review by bsekachev — APPROVED** [review]

File: cvat/apps/events/signals.py
Comment (nmanovic) [inline cvat/apps/events/signals.py:24]: should we specify dispatch_uuid here for every receiver?

File: cvat/apps/events/serializers.py
Comment (nmanovic) [inline cvat/apps/events/serializers.py:23]: Let's add user_name and user_email. Thus it will be easy to look for a user in ClickHouse and Grafana.

File: cvat/apps/events/serializers.py
Comment (nmanovic) [inline cvat/apps/events/serializers.py:24]: Let's add org_slug

**Review by nmanovic — COMMENTED** [review]
Please look for `kibana` word and cleanup

File: components/analytics/grafana/dashboards/Untitled-1.sql
Comment (nmanovic) [inline components/analytics/grafana/dashboards/Untitled-1.sql:1]: Is it possible to rename the file?

File: cvat-core/src/session-implementation.ts
Comment (nmanovic) [inline cvat-core/src/session-implementation.ts:811]: project_id?

Comment (nmanovic) [reply cvat-core/src/session-implementation.ts:811]: task_id?

File: cvat/apps/engine/tests/test_rest_api.py
Comment (nmanovic) [inline cvat/apps/engine/tests/test_rest_api.py:527]: probably the name should be unified with UI.

File: components/analytics/grafana/dashboards/Untitled-1.sql
Comment (azhavoro) [reply components/analytics/grafana/dashboards/Untitled-1.sql:1]: this is tmp file, I will remove it

File: cvat/apps/events/export.py
Comment (nmanovic) [inline cvat/apps/events/export.py:69]: Why IMPORT_DATA queue?

File: cvat/settings/base.py
Comment (nmanovic) [inline cvat/settings/base.py:221]: We will need to remove the extension. It isn't well supported. Let's think how to do that.

File: cvat/apps/events/handlers.py
Comment (nmanovic) [inline cvat/apps/events/handlers.py:68]: Why do we need to call _get_current_user? We handle a request. The request has an authentication user. Why should we get a user from job.segment.task.owner?

**Review by nmanovic — APPROVED** [review]
Let's merge the PR. I will propose to discuss some tricky moments on Monday.

**Pull request merged**
